src/display/api.js

```diff
@@ -1091,6 +1091,17 @@ class PDFDocumentProxy {
   getCalculationOrderIds() {
     return this._transport.getCalculationOrderIds();
   }
+
+  /**
+   * For testing purposes.
+   * @returns {Promise<Object> | null} A XML tree for the xfa:datasets.
+   */
+  getXFADatasets() {
+    if (typeof PDFJSDev === "undefined" || PDFJSDev.test("TESTING")) {
+      return this._transport.getXFADatasets();
+    }
+    return null;
+  }
```

Comment:
You might be able to limit this method to *only* the builds where it makes sense, by instead adding something like the following at the end of the `PDFDocumentProxy`-constructor:
```javascript
if (typeof PDFJSDev === "undefined" || PDFJSDev.test("TESTING")) {
  Object.defineProperty(this, "getXFADatasets", {
    value: () => {
      return this._transport.getXFADatasets();
    },
  });
}
```